### PR TITLE
Viz markers backwards compatible

### PR DESCRIPTION
**Pull request by salhus**
This PR updates the functionality in PR #706 .

Corresponding documentation has been updated. Examples have been updated in Applications repo with corresponding documentation.
- Updated the initialization such that if user defines the array of marker coordinates, visualization is initialized,
- If the user does not define the coordinates as an Nx2 array, the user will get the following error message.

image 

Here is an example of initializing the visualization blocks. The examples are updated in the Applications Repo.
Examples:
1. An example with a square mesh of visualization blocks:
For :code:
``mrk = 10;``
``dmrk = 5;``
``[X,Y] = meshgrid(imagemrk:dmrk:mrk,-mrk:dmrk:mrk);``
``waves.markLoc = [reshape(X,[],1),reshape(Y,[],1)]; % Marker Locations [X,Y]``
``clear('mrk','dmrk','X','Y')``

``waves.markStyle   = 3;    % 1: Sphere, 2: Cube, 3: Frame.``
``waves.markSize    = 20;   % Marker Size in Pixels.``

### CODE REVIEW

**Comment by salhus** [comment]
@nathanmtom @kmruehl 

here is the updated PR with enable/disable feature.

**Comment by salhus** [comment]
@nathanmtom  
As discussed on Friday during open-office hours. I updated the PR to have,
- Remove the initialization using a variable. Now the user only needs to define the Nx2 array,
- If the array dimensions are incorrect the user will get an error message,
- Updated the applications examples to have @jtgrasb  's visualization in the user-defined-functions file,
- Updated the corresponding docs.

**Comment by akeeste** [comment]
@salhus @nathanmtom Other than the small waveClass conflict, is this PR close to merging? I have a couple of library PRs to bring in, one I'd like to do one top of these changes.

File: source/functions/initializeWecSim.m [hunk 1]
```diff
@@ -382,6 +382,17 @@
 %    eval(['sv_b' num2str(ii) '_rigidBody = Simulink.Variant(''nhbody_' num2str(ii) '==0'');'])
 end; clear ii
 
+% Visualization Blocks
+
+if ~isempty(waves.waveAmpTimeViz)
```
Comment (nathanmtom) [inline source/functions/initializeWecSim.m:387]: @salhus Based on your last comment on this PR you were removing this variable; however, when I pull your latest files this is still present. Did you push you latest changes?

File: source/functions/stopWecSim.m [hunk 2]
```diff
@@ -8,6 +8,7 @@
 % Clear intermediate variables and variant subsystem flags
 clear nlHydro sv_linearHydro sv_nonlinearHydro ssCalc radiation_option sv_convolution sv_stateSpace sv_constantCoeff typeNum B2B sv_B2B sv_noB2B;
 clear nhbod* sv_b* sv_noWave sv_regularWaves sv_irregularWaves sv_udfWaves sv_instFS sv_meanFS sv_MEOn sv_MEOff morisonElement flexHydrobody_* sv_irregularWavesNonLinYaw sv_regularWavesNonLinYaw yawNonLin numBody;
+clear sv_visualizationOFF sv_visualizationON visON X Y
```
Comment (nathanmtom) [inline source/functions/stopWecSim.m:11]: @salhus the visOn variable still seems to be included when you mentioned removing.

**Review by nathanmtom — APPROVED** [review]
@salhus Apologies as I needed to look through the file changes a bit more in-depth to understand what is happening. But, once the conflict with `waveClass` is resolved then this pull request should be good to go. After resolving the conflict let's make sure that we are passing all checks again.

**Comment by akeeste** [comment]
@nathanmtom @salhus I was familiar with the conflict (it was with a previous update I made to the wave setup) so I went ahead and resolved it. All tests are still passing after combining the waveClass checks

**Comment by salhus** [comment]
@akeeste  Thanks so much for resolving that !

**Comment by nathanmtom** [comment]
@akeeste Thanks for resolving the conflict. 

@salhus Can you run the WECSimApplications case test file with this pull request? If all checks pass (except the moordyn and paraview checks if you do not have those installed) then I think this can be merged into Dev. Since this feature does not change any force or dynamics calculation we shouldn't have to worry about it affecting simulation outputs.

**Comment by salhus** [comment]
> @akeeste Thanks for resolving the conflict.
> 
> @salhus Can you run the WECSimApplications case test file with this pull request? If all checks pass (except the moordyn and paraview checks if you do not have those installed) then I think this can be merged into Dev. Since this feature does not change any force or dynamics calculation we shouldn't have to worry about it affecting simulation outputs.

@nathanmtom I ran the tests. It is failing just one test, the WECCOMP. (Attaching the summary)
[testsAppRepo.txt](https://github.com/WEC-Sim/WEC-Sim/files/7246306/testsAppRepo.txt)

**Comment by kmruehl** [comment]
@salhus @akeeste and @nathanmtom thanks for working on this. I'm going to review it now.

File: docs/user/advanced_features.rst [hunk 3]
```diff
@@ -830,6 +831,66 @@ of the 1st body. The position, velocity and acceleration of that body is shown.
...
+``[X,Y] = meshgrid(-mrk:dmrk:mrk,-mrk:dmrk:mrk);``
+``waves.markLoc = [reshape(X,[],1),reshape(Y,[],1)]; % Marker Locations [X,Y]``
+``clear('mrk','dmrk','X','Y')``
+
+``waves.markStyle   = 3;    % 1: Sphere, 2: Cube, 3: Frame.``
+``waves.markSize    = 20;   % Marker Size in Pixels.``
+
+.. Note:: 
+
+    The `WEC-Sim Applications <https://github.com/WEC-Sim/WEC-Sim_Applications>`_ repository has a demonstration for this approach.
```
[the remark this answers is not in the document]
Comment (kmruehl) [reply docs/user/advanced_features.rst]: This relates to https://github.com/WEC-Sim/WEC-Sim_Applications/pull/12

**Comment by kmruehl** [comment]
To Do:
- [ ] clean up variable naming
- [ ] move Frames to sub-library
- [ ] update documentation

**Review by kmruehl — APPROVED** [review]
@salhus great job on this PR, thank you! I just have two minor questions (above) that we may want to resolve prior to a merge

**Comment by kmruehl** [comment]
All wec-sim and wec-sim_application tests are passing.

**Comment by salhus** [comment]
@kmruehl 
Thanks for the review. I had the dialog boxes as a way to know what settings were selected by the user. It also helped me do the mask initialization when doing the initialization for the variable subsytems without using a 'comment/uncomment'  code which older matlabs didnt support.
Let me know if that answers the question..

**Pull request merged**